### DISCUSSION TITLE
How to add cache from local tarball with a specified algorithm?

### DISCUSSION THREAD

**Discussion by bekcpear** [npm/feedback#547] [CLI] [answered]
I want pack a npm package to gentoo linux. There is no internet connection in compile/install phase of portage by default, so I need to download all dependencies first and add to a sandbox environment in the form of cache.

However, there are many dependencies only with a sha1 algorithm, the default algorithm used by `npm cache add` for **local tarball** is sha512, which make the `npm ci` failed.

I spent a lot of time looking for a solution but failed. 'npm/pacote' can specify algorithm when fetching tarball, but npm cli does not provide a method to change default algorithm.

The workaround I can think of is to create a new package-lock.json file with all sha512 integrity metadata, however it's not a good method.

So, can I make a request for a new feature that can specify an algorithm while executing `npm cache add ...` . I think it's useful.

**Comment by ljharb** [comment]
Can you tarball an installed node_modules folder?

**Reply by bekcpear** [reply]
Yes I can, but I want to try to avoid this way of redistributing all dependencies.

**Reply by ljharb** [reply]
Why? That’s how it’s done.

**Comment by MylesBorins** [comment, marked as answer]
@bekcpear if you would like to suggest a new feature for the CLI the next step would be to open an issue over at https://github.com/npm/rfcs to discuss the feature you would like to add
